curve25519-dalek/src/edwards.rs

```diff
@@ -574,12 +576,37 @@ impl EdwardsPoint {
         CompressedEdwardsY(s)
     }
 
+    #[cfg(feature = "digest")]
+    /// Perform hashing to curve, with explicit hash function and DST using the suite
+    /// edwards25519_XMD:SHA-512_ELL2_NU_
+    ///
+    /// See https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-12#section-6.8.2
+    pub fn hash_to_curve_dst<D>(bytes: &[u8], dst: &[u8]) -> EdwardsPoint
```

Comment:
Can we rename `dst` to `domain_sep` or something similar, and update the docstring? `dst` reads too much like "destination" imo

Reply:
We could also simply cal it `hash_to_curve`, given that that's how the spec is defined.